pkg/client/client.go

```diff
@@ -137,6 +138,25 @@ func (c *Client) ServerAPIVersions() (*api.APIVersions, error) {
 	return &v, nil
 }
 
+type ClusterValidationStateInterface interface {
+	ClusterValidationState() (*api.ServerStatusList, error)
+}
+
+// ClusterValidationState retrieves and parses the server's self-monitored cluster state.
+func (c *Client) ClusterValidationState() (*api.ServerStatusList, error) {
+	body, err := c.Get().AbsPath("/validate").Do().Raw()
```

Comment:
Yes, I discovered a bug where `Do().Raw()` does the wrong thing when the non-`apiStatus` JSON response includes a `status` field as is the case with Elasticsearch. I refactored it into two distinct phases which first processes the response without assuming it is an `api.Status` (this is done by `DoRaw()`) and then interpret it as an `api.Status` which is done by `Do()` which now calls `DoRaw()`. I think we should replace all occurrences of `Do().Raw()` with `DoRaw()` and remove the `Raw()` method.